### PR TITLE
1905 ccd async

### PR DESCRIPTION
**Pull request by RamilGaripov**
Ticket: https://github.com/metriport/metriport-internal/issues/1905

### Dependencies
- Upstream: https://github.com/metriport/metriport/pull/2332

### Description
- Making the CCD gen on DQ async
- In the meantime, quickly generating an empty CCD

### Testing
- Local
  - [ ] Test generation on patient without contributed data
  - [ ] Test generation on patient with contributed data (see that empty is generated before real one)

### Release Plan
- [ ] Merge this

### CODE REVIEW

File: packages/core/src/external/aws/s3.ts [hunk 1]
```diff
@@ -356,41 +356,6 @@ export class S3Utils {
     }
   }
 
-  async retrieveDocumentIdsFromS3(
```
Comment (RamilGaripov) [inline packages/core/src/external/aws/s3.ts:359]: Moving this function out of `external/aws/s3`, as discussed here - [context](https://metriport.slack.com/archives/C04DMKE9DME/p1719274157562819?thread_ts=1718897333.131819&cid=C04DMKE9DME)

File: packages/core/src/external/carequality/dq/process-inbound-dq.ts [hunk 2]
```diff
@@ -80,3 +87,39 @@ async function getDocumentContents(cxId: string, patientId: string): Promise<str
   }
   return documentContents;
 }
+
+async function retrieveXmlContentsFromMetadataFilesOnS3(
```
Comment (RamilGaripov) [inline packages/core/src/external/carequality/dq/process-inbound-dq.ts:91]: Put the function here. Renamed it to better match its function

File: packages/core/src/external/carequality/dq/process-inbound-dq.ts [hunk 3]
```diff
@@ -80,3 +87,39 @@ async function getDocumentContents(cxId: string, patientId: string): Promise<str
...
+  cxId: string,
+  patientId: string,
+  bucketName: string
+): Promise<string[]> {
+  const folderName = createFolderName(cxId, patientId);
+  const Prefix = `${folderName}/${UPLOADS_FOLDER}/`;
```
Comment (RamilGaripov) [inline packages/core/src/external/carequality/dq/process-inbound-dq.ts:97]: Making use of existing functions and constants

File: packages/core/src/external/carequality/dq/process-inbound-dq.ts [hunk 4]
```diff
@@ -37,15 +38,21 @@ export async function processInboundDocumentQuery(
...
       const queryParams = {
         cxId,
         patientId,
       };
       const params = new URLSearchParams(queryParams).toString();
-      const endpointUrl = `${apiUrl}/internal/docs/ccd`;
-      const url = `${endpointUrl}?${params}`;
-      await executeWithNetworkRetries(async () => await api.post(url), { log });
+
+      executeWithNetworkRetries(async () => api.post(`${apiUrl}/internal/docs/ccd?${params}`), {
```
Comment (jonahkaye) [inline packages/core/src/external/carequality/dq/process-inbound-dq.ts:48]: intentional? why is this unawaited and duplicated here and below?

Comment (RamilGaripov) [reply packages/core/src/external/carequality/dq/process-inbound-dq.ts:48]: There was a discussion regarding this [here](https://metriport.slack.com/archives/C04DMKE9DME/p1719270581196869?thread_ts=1718897333.131819&cid=C04DMKE9DME). 

At this step, we're triggering generation of the full CCD. And in the meantime, we're calling another endpoint to very quickly generate an empty one

Comment (RamilGaripov) [reply packages/core/src/external/carequality/dq/process-inbound-dq.ts:48]: So this is completely intentional

**Review by jonahkaye — APPROVED** [review]

**Pull request merged**